Add test_physical_discovery to dual Galaxy workflow

jpanasiukTT opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/30272

### Problem description
As described in the issue.

### What's changed
TODO

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

tt-asaigal reviewed (commented):
Thanks for the changes @jpanasiukTT.

I think that we can reduce the number of tests added overall while ensuring that we maintain coverage.

Please see the comments below for more details, but high level I'm thinking that:
- We have 3 sanity unicast tests:
    - Mesh 0 to Mesh 1 single sender
    - Mesh 1 to Mesh 0 single sender
    - Mesh 0 to Mesh 1 multiple senders (maybe randomly sampled up to 4)
- We have 1 stress test inducing unidirectional (Mesh 0 to Mesh 1) traffic. High packet count and can sweep over the packet size

The purpose of the first 3 is to ensure that changes to the fabric routers don't cause basic regressions.

The purpose of the stress test is to expose any latent instabilities or hard to find issues exposed by new changes.

**Note to reviewers:** We're currently hardcoding our traffic patterns, since randomized tests or those with top level patterns can easily expose known deadlocks for multi-mesh setups.

jpanasiukTT committed:
Add dual Galaxy multi-mesh configuration and test suite

- Create dual_galaxy_multi_mesh_mgd.yaml mesh descriptor with 2 separate
  8x4 meshes connected via 8 relaxed ethernet links
- Add test_dual_galaxy_multi_mesh.yaml with comprehensive fabric tests:
  - Single sender/receiver unicast (forward and reverse directions)
  - Multiple senders unicast
  - Multicast stress test
  - Point-to-point unidirectional
- All tests use unidirectional traffic patterns to avoid deadlocks

jpanasiukTT committed:
Integrate multi-mesh tests into dual Galaxy test suite

Add test_physical_discovery, run_cluster_validation, and multi-mesh
fabric tests to the dual Galaxy test runner to validate:
- Physical device discovery across meshes
- Cluster connectivity and validation
- Multi-mesh fabric traffic patterns

jpanasiukTT committed:
Enhance dual Galaxy testing with multiple link configurations

jpanasiukTT committed:
Merge branch 'main' into jpanasiuk/enhance-dual-galaxy-testing

jpanasiukTT committed:
change mesh desc paths

jpanasiukTT committed:
reinstate mpirun-ulfm cmds in run_dual_galaxy_tests.sh

jpanasiukTT committed:
make traffic pattern relaxed
